ui-v2/app/instance-initializers/selection.js

```diff
@@ -0,0 +1,49 @@
+const isSelecting = function(win = window) {
+  const selection = win.getSelection();
+  let selecting = false;
+  try {
+    selecting = 'isCollapsed' in selection && !selection.isCollapsed;
+  } catch (e) {
+    // passthrough
+  }
+  return selecting;
+};
+export default {
+  name: 'selection',
+  initialize(container) {
+    const dom = container.lookup('service:dom');
+    const findAnchor = function(el) {
+      return el.tagName === 'A' ? el : dom.closest('a', el);
+    };
+    const mousedown = function(e) {
+      const $a = findAnchor(e.target);
+      if ($a) {
+        const href = $a.getAttribute('href');
+        if (href) {
+          $a.dataset.href = href;
+          $a.removeAttribute('href');
```

Comment:
> What happens if between the mousedown and the mouseup something (e.g., a blocking query) causes a re-render?

If the blocking query resolves while I'm 'mid-click', then the `href` will be reset, and thinking about it (although not entirely sure), you would navigate to the `href` (using the `data-href`) that was previously there. If you were just clicking as a normal user though (with a split second click), this is 'probably' what is expected. As a user I probably didn't even notice the rendering change, I probably wanted to go to the service that I mouse down'ed on.

If I click and hold for a while, and then a blocking query updates, which by chance changes the `href`, whilst very edge casey but I suppose is possible 🤷‍♂ , it would probably take you to the old link - I'll take a look at that later to be sure. But, I don't think this would make me want to back out of this. It's extremely edge casey - and I'm not sure in what usecase somebody might be clicking and holding - apart from selecting text, in which case the href would get updated to the new value when you next click to navigate.

I think another thing that could make me back out is... If this breaks right click/open in new tab, which it might, so I'm not going to merge this until I've looked at that. I'm guessing that's a straightforward fix if that's a problem.